plugins/content-manager/src/main/java/com/wazuh/contentmanager/cti/catalog/client/ApiClient.java

```diff
@@ -115,4 +116,34 @@ public SimpleHttpResponse getConsumer(String context, String consumer) throws Ex
 
         return future.get(PluginSettings.getInstance().getClientTimeout(), TimeUnit.SECONDS);
     }
+
+    /**
+     * Retrieves the changes for a specific consumer within a given context.
+     *
+     * @param context    The context identifier.
+     * @param consumer   The consumer identifier.
+     * @param fromOffset The starting offset (exclusive).
+     * @param toOffset   The ending offset (inclusive).
+     * @return A {@link SimpleHttpResponse} containing the API response.
+     * @throws ExecutionException   If the computation threw an exception.
+     * @throws InterruptedException If the current thread was interrupted while waiting.
+     * @throws TimeoutException     If the wait timed out.
+     */
+    public SimpleHttpResponse getChanges(String context, String consumer, long fromOffset, long toOffset) throws ExecutionException, InterruptedException, TimeoutException {
+        String uri = this.buildConsumerURI(context, consumer) + "/changes?from_offset=" + fromOffset + "&to_offset=" + toOffset;
+
+        SimpleHttpRequest request = SimpleRequestBuilder
+            .get(uri)
+            .build();
+
+        final Future<SimpleHttpResponse> future = client.execute(
+            SimpleRequestProducer.create(request),
+            SimpleResponseConsumer.create(),
+            new HttpResponseCallback(
+                request, "Outgoing request failed"
```

Comment:
The error message could be more informative, something like 
```suggestion
                request, "Failed to send request to CTI service"
```
Im not sure if we can get the error, or error code, that would be great